compiler/src/regalloc.ml

```diff
@@ -92,10 +90,12 @@ let find_equality_constraints (id: instruction_desc) : arg_position list list =
 
 let find_var outs ins ap : _ option =
   match ap with
-  | APout n -> (List.nth outs n |> function Lvar v -> Some v | _ -> None)
+  | APout n ->
+     Option.bind (Oseq.onth outs n)
+       (function Lvar v -> Some v | _ -> None)
   | APin n ->
-     (List.nth ins n |>
-        function
+     Option.bind (Oseq.onth ins n)
+       (function
```

Comment:
Why do you use options there? If `outs` or `ins` are not long enough, I think this means there is an error in the instruction description, which we would like to report rather than ignore?

Reply:
You’re right. Let me fix that.

Reply:
Done. The error message is not very informative.

Reply:
In general, our internal errors are not very informative, that's ok.